### PR TITLE
fix(scanner): handle digest-only MainAssetName without panic

### PR DESCRIPTION
**Pull request by henrikfrech**
## What
Fix panic in scanner report conversion when Sysdig returns digest-only references in `mainAssetName` (e.g. `repo@sha256:...`).

## Why
`ToHarborVulnerabilityReport` assumed `repo:tag@digest` and sliced with `lastColon=-1` for tagless images, causing:
`panic: runtime error: slice bounds out of range [:-1]`.

## Changes
- Added robust `parseMainAssetName` helper in `pkg/scanner/base_adapter.go`
- Supports:
  - tagged: `repo:tag@sha256:...`
  - digest-only: `repo@sha256:...`
  - registry-prefixed variants
- On malformed values, logs warning and continues instead of panicking
- Added parser-focused tests in `pkg/scanner/base_adapter_test.go`
- Updated scanner test fixture expectation in `pkg/scanner/backend_adapter_test.go`

## Validation
- `go test -count=1 ./pkg/scanner` ✅
- `go test ./pkg/scanner/...` ✅

## Impact
- Unblocks Harbor report generation for digest-only image references
- Preserves existing tagged-image behavior

### CODE REVIEW

File: pkg/scanner/base_adapter.go [hunk 1]
```diff
@@ -117,6 +115,38 @@ func (b *BaseAdapter) ToHarborVulnerabilityReport(repository string, shaDigest s
...
+
+	namePart := parts[0]
+	hash = parts[1]
+
+	lastSlash := strings.LastIndex(namePart, "/")
+	lastColon := strings.LastIndex(namePart, ":")
+	if lastColon > lastSlash {
+		tag = namePart[lastColon+1:]
+		namePart = namePart[:lastColon]
+	}
```
[the remark this answers is not in the document]
Comment (airadier) [reply pkg/scanner/base_adapter.go:150]: Fixed in 22d3b61. `parseMainAssetName` now returns `ok=false` when a tag separator is present but the tag is empty (e.g. `repo:@sha256:...`), so the caller logs a warning and skips the entry consistently.

File: pkg/scanner/base_adapter_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		Expect(hash).To(Equal("sha256:abc"))
+	})
+
+	It("returns not ok for malformed values", func() {
+		for _, tc := range []string{
+			"",
+			"@sha256:abc",
+			"sysdig/agent@",
+			"sysdig/agent",
+			"sysdig/agent:1.0",
```
[the remark this answers is not in the document]
Comment (airadier) [reply pkg/scanner/base_adapter_test.go:98]: Added in 22d3b61. The empty-tag form `sysdig/agent:@sha256:abc` is now included in the malformed-values test table, locking in the `ok=false` behavior.

**Review by airadier — APPROVED** [review]
Approving. 

This fixes a real, reproducible panic (`slice bounds out of range [:-1]`) in `ToHarborVulnerabilityReport` when Sysdig returns a digest-only `MainAssetName`, which is hit on the active InlineAdapter path (`inline_adapter.go:248`). As a bonus it also fixes a latent bug where un-prefixed repo names (e.g. `sysdig/agent`) had their first segment stripped as if it were a registry, so `Artifact` was silently never populated — hence the corrected fixture in `backend_adapter_test.go`.

Reviewed via 3 independent review passes (Go idioms, general quality, adversarial edge cases): unanimous, 0 critical/high. The parser was exercised against 30+ reference forms with no panics, and the added tests lock in the tricky cases (digest-only, registry+port disambiguation, empty tag).

The two Copilot comments (empty-tag rejection + its test) are addressed, and I pushed two small maintainability commits (doc comment on `parseMainAssetName`, `hash`→`digest` rename). Remaining low-severity items are pre-existing and out of scope — tracked as a follow-up issue.

LGTM. Thanks @henrikfrech for the fix.
